app/services/status_update_service.rb

```diff
@@ -1,5 +1,7 @@
 class StatusUpdateService
-  def initialize(reference:, status:, user: nil)
+  def initialize(sent_at:, completed_at:, reference:, status:, user: nil)
```

Comment:
Just personal opinion I guess, but I think `sent_at` and `completed_at` should appear after the `reference` and `status` arguments. I definitely think the `reference` should come first as it's the delivery attempt that this service class acts on.